self_unenrollment_tool/views.py

```diff
@@ -0,0 +1,245 @@
+
+from logging import getLogger
+
+from canvas_sdk import RequestContext
+from canvas_sdk.methods.enrollments import (conclude_enrollment,
+                                            list_enrollments_courses)
+from django.conf import settings
+from django.contrib import messages
+from django.http import HttpRequest, JsonResponse
+from django.shortcuts import redirect, render
+from django.urls import reverse
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+from icommons_common.models import CourseEnrollee, CourseGuest, CourseStaff
+from pylti1p3.contrib.django import (DjangoCacheDataStorage, DjangoDbToolConf,
+                                     DjangoMessageLaunch, DjangoOIDCLogin)
+
+from self_enrollment_tool.models import SelfEnrollmentCourse
+
+from .lti1p3_utils import (get_launch_url, get_message_launch,
+                           require_lti_launch)
+
+logger = getLogger(__name__)
+
+SDK_SETTINGS = settings.CANVAS_SDK_SETTINGS
+# make sure the session_inactivity_expiration_time_secs key isn't in the settings dict
+SDK_SETTINGS.pop('session_inactivity_expiration_time_secs', None)
+SDK_CONTEXT = RequestContext(**SDK_SETTINGS)
+
+
+@csrf_exempt
+def login(request: HttpRequest):
+    tool_conf = DjangoDbToolConf()
+    launch_data_storage = DjangoCacheDataStorage()
+
+    oidc_login = DjangoOIDCLogin(request, tool_conf, launch_data_storage=launch_data_storage)
+    target_link_uri = get_launch_url(request)
+    return oidc_login.enable_check_cookies().redirect(target_link_uri)
+
+
+@require_POST
+@csrf_exempt
+def launch(request: HttpRequest):
+    message_launch = get_message_launch(request)
+    # this next line is necessary even if the variable is not used; if get_launch_data() is not called, launch data will not be saved to the session!
+    launch_data = message_launch.get_launch_data()
+    logger.info('self_unenrollment_tool launch', extra={'launch_data': launch_data})
+    return redirect(reverse('self_unenrollment_tool:index', kwargs={'launch_id': message_launch.get_launch_id()}))
+
+
+@require_lti_launch
+def index(request: HttpRequest, launch_id: str):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 55 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this function or break it into multiple functions.

Reply:
I've split out the GET and POST logic into separate functions.